checker/jdk/signedness/src/java/lang/Integer.java

```diff
@@ -0,0 +1,1602 @@
+
+    /**
+     * The number of bytes used to represent a {@code int} value in two's
+     * complement binary form.
+     *
+     * @since 1.8
+     */
+    public static final @Positive int BYTES = SIZE / Byte.SIZE;
+
+    /**
+     * Returns an {@code int} value with at most a single one-bit, in the
+     * position of the highest-order ("leftmost") one-bit in the specified
+     * {@code int} value.  Returns zero if the specified value has no
+     * one-bits in its two's complement binary representation, that is, if it
+     * is equal to zero.
+     *
+     * @param i the value whose highest one bit is to be computed
+     * @return an {@code int} value with a single one-bit, in the position
+     *     of the highest-order one-bit in the specified value, or zero if
+     *     the specified value is itself equal to zero.
+     * @since 1.5
+     */
+    public static int highestOneBit(@PolySigned int i) {
+        // HD, Figure 3-1
+        i |= (i >>  1);
+        i |= (i >>  2);
+        i |= (i >>  4);
+        i |= (i >>  8);
+        i |= (i >> 16);
+        return i - (i >>> 1);
+    }
+
+    /**
+     * Returns an {@code int} value with at most a single one-bit, in the
+     * position of the lowest-order ("rightmost") one-bit in the specified
+     * {@code int} value.  Returns zero if the specified value has no
+     * one-bits in its two's complement binary representation, that is, if it
+     * is equal to zero.
+     *
+     * @param i the value whose lowest one bit is to be computed
+     * @return an {@code int} value with a single one-bit, in the position
+     *     of the lowest-order one-bit in the specified value, or zero if
+     *     the specified value is itself equal to zero.
+     * @since 1.5
+     */
+    public static int lowestOneBit(@PolySigned  int i) {
+        // HD, Section 2-1
+        return i & -i;
+    }
+
+    /**
+     * Returns the number of zero bits preceding the highest-order
+     * ("leftmost") one-bit in the two's complement binary representation
+     * of the specified {@code int} value.  Returns 32 if the
+     * specified value has no one-bits in its two's complement representation,
+     * in other words if it is equal to zero.
+     *
+     * <p>Note that this method is closely related to the logarithm base 2.
+     * For all positive {@code int} values x:
+     * <ul>
+     * <li>floor(log<sub>2</sub>(x)) = {@code 31 - numberOfLeadingZeros(x)}
+     * <li>ceil(log<sub>2</sub>(x)) = {@code 32 - numberOfLeadingZeros(x - 1)}
+     * </ul>
+     *
+     * @param i the value whose number of leading zeros is to be computed
+     * @return the number of zero bits preceding the highest-order
+     *     ("leftmost") one-bit in the two's complement binary representation
+     *     of the specified {@code int} value, or 32 if the value
+     *     is equal to zero.
+     * @since 1.5
+     */
+    public static @NonNegative int numberOfLeadingZeros(@UnknownSignedness int i) {
```

Comment:
There is an inconsistency between numberOfLeadingZeros and numberOfLeadingZeros.  Both can be called on any argument, but one uses `@UnknownSignedness` whereas the other uses `@PolySigned`.  If the argument's type isn't used elsewhere in the signature, I suggest using `@UnknownSignedness` consistently throughout.  (An exception is for argument element types, such as in `String` constructors; there you need to use `@PolySigned` even though no other type in the signature is polymorphic.)

Reply:
@mernst I am unable to understand this statement. Please, can you explain!

Reply:
This is not a helpful question, because it is too vague.
Which statement do you not understand?
All 4 sentences that I wrote, or just some of them?
What *do* you understand, and what *don't* you understand?

Reply:
> There is an inconsistency between numberOfLeadingZeros and numberOfLeadingZeros. 

I did not understood this statement.

Reply:
> Both can be called on any argument, but one uses @UnknownSignedness whereas the other uses @PolySigned

Why don't you annotate both methods with the same annotation?  Is there a reason to prefer different annotations for the two methods?

Reply:
@mernst About which two methods you are talking about? You have written numberOfLeadingZeros twice. Please rectify that.
>There is an inconsistency between numberOfLeadingZeros and numberOfLeadingZeros.

Reply:
highestOneBit